src/main/java/com/fzi/externalcontrol/impl/RequestProgram.java

```diff
@@ -89,4 +109,36 @@ public String sendCommand(String command) {
     }
     return result;
   }
+
+  public void splitProgram(String program) {
+    this.header = "";
+    this.controlLoop = "";
+    Boolean headerEndFound = false;
+    Boolean headerBeginFound = false;
+    String[] programArray = program.split("\n");
+
+    for (int it = 0; it < programArray.length; ++it) {
+      if (programArray[it].matches("(.*)# HEADER_BEGIN(.*)")) {
```

Comment:
Allowing only whitespaces before and after might be safer. If people write comments like

```python
# This does not go into the "# HEADER_BEGIN" section
```

```suggestion
      if (programArray[it].matches("(\s*)# HEADER_BEGIN(\s*)")) {
```

Reply:
This seems to brake the URCap when building, to make it build following `if (programArray[it].matches("(\\s*)# HEADER_BEGIN(\\s*)"))` will be needed. 

When testing it with `# This does not go into the "# HEADER_BEGIN" section` the only time it seems to be able to find the `# HEADER_BEGIN` is with the original implementation.

Reply:
According to [this post](https://stackoverflow.com/questions/4731055/whitespace-matching-regex-java) `\\p{Zs}` might be a working solution.

Reply:
I tested `\\p{Zs}` and it recognizes `# HEADER_BEGIN` but not `# This is a comment about # HEADER_BEGIN`

@urmahp If you would be fine with this, I will push that to your branch and then we're ready to merge.

Reply:
Yes I have also tested that it works and it seems like a good solution, thanks for you help.